pkg/apis/networking/validation/validation.go

```diff
@@ -59,20 +61,39 @@ func ValidateNetworkPolicyName(name string, prefix bool) []string {
 
 // ValidateNetworkPolicyPort validates a NetworkPolicyPort
 func ValidateNetworkPolicyPort(port *networking.NetworkPolicyPort, portPath *field.Path) field.ErrorList {
+
+	allowEndPort := utilfeature.DefaultFeatureGate.Enabled(features.NetworkPolicyEndPort)
+
 	allErrs := field.ErrorList{}
 	if port.Protocol != nil && *port.Protocol != api.ProtocolTCP && *port.Protocol != api.ProtocolUDP && *port.Protocol != api.ProtocolSCTP {
 		allErrs = append(allErrs, field.NotSupported(portPath.Child("protocol"), *port.Protocol, []string{string(api.ProtocolTCP), string(api.ProtocolUDP), string(api.ProtocolSCTP)}))
 	}
+
+	if !allowEndPort && port.EndPort != nil {
+		allErrs = append(allErrs, field.Forbidden(portPath.Child("endPort"), "This field can be enabled with the NetworkPolicyEndPort feature gate"))
+	}
 	if port.Port != nil {
 		if port.Port.Type == intstr.Int {
 			for _, msg := range validation.IsValidPortNum(int(port.Port.IntVal)) {
 				allErrs = append(allErrs, field.Invalid(portPath.Child("port"), port.Port.IntVal, msg))
 			}
+			if allowEndPort && *port.EndPort < port.Port.IntVal {
+				// TODO: Find the right error field
+				allErrs = append(allErrs, field.Invalid(portPath.Child("endPort"), port.Port.IntVal, "endPort field must be equal or bigger than port field"))
+			}
 		} else {
+			if allowEndPort && port.EndPort != nil {
+				// TODO: Find the right error field
+				allErrs = append(allErrs, field.Invalid(portPath.Child("endPort"), port.Port.IntVal, "endPort cannot be used when port is non-numeric"))
+			}
 			for _, msg := range validation.IsValidPortName(port.Port.StrVal) {
 				allErrs = append(allErrs, field.Invalid(portPath.Child("port"), port.Port.StrVal, msg))
 			}
 		}
+	} else {
+		if allowEndPort && port.EndPort != nil {
+			allErrs = append(allErrs, field.Invalid(portPath.Child("endPort"), port.Port.StrVal, "endPort cannot be defined when port is not defined"))
```

Comment:
`port.Port` is `nil` here

Reply:
OK, changed to *port.EndPort (actually this is the field that matters)